tutor/commands/dev.py

```diff
@@ -45,6 +48,45 @@ def dev(context: click.Context) -> None:
     context.obj = DevContext(context.obj.root)
 
 
+def parse_ports(docker_ps_output: str) -> list[int]:
+    """
+    Extracts the ports from the docker ps output in json format.
+    """
+    exposed_ports = []
+    for line in docker_ps_output.splitlines():
+        publishers = json.loads(line)["Publishers"]
+        for publisher in publishers:
+            port = publisher["PublishedPort"]
+            if port:
+                exposed_ports.append(port)
+    return exposed_ports
+
+
+@click.command(help="List the status of all services.")
+@click.pass_obj
+def hosts(context: compose.BaseComposeContext) -> None:
+    context_name = context.NAME
```

Comment:
this variable is only used once, we can reference it directly instead of creating a variable.